### PR TITLE
Fix/cs 43301 merge live preview

### PR DESCRIPTION
**Pull request by abhinav-from-contentstack**
- merged changes from #100, resolved merge conflicts
- added changes to webpack config for web build

### CODE REVIEW

**Review by harshithad0703 — APPROVED** [review]
Approved. Thank you

File: config.js [hunk 1]
```diff
@@ -13,7 +13,7 @@ const config = {
     },
     live_preview: {
         enable: false,
-        host: 'api.contentstack.io'
+        host: 'rest-preview.contentstack.com'
```
Comment (ishaileshmishra) [inline config.js:16]: @harshithad0703 Please confirm the host. looks like we are adding some incorrect host here.

Comment (abhinav-from-contentstack) [reply config.js:16]: @Deepak-Kharah , would you confirm if this host is correct?

File: src/core/stack.js [hunk 2]
```diff
@@ -73,6 +73,10 @@ export default class Stack {
 
         if(stack_arguments[0].region && stack_arguments[0].region !== undefined && stack_arguments[0].region !== "us") {
             this.config['host'] = stack_arguments[0].region+"-"+"cdn.contentstack.com";
+            this.config["live_preview"]["host"] =
+                stack_arguments[0].region +
+                "-" +
+                "rest-preview.contentstack.com";
```
Comment (ishaileshmishra) [inline src/core/stack.js:79]: If it's hard coded, we can write somewhere and use it multiple places. @harshithad0703 

Like it used in 
config as well as stack

File: config.js (hunk 1 above)
Comment (Deepak-Kharah) [reply config.js:16]: Yes, the host is correct

Comment (ishaileshmishra) [reply config.js:16]: Hi I double check the host. It's `rest-preview`

**Review by Deepak-Kharah — APPROVED** [review]
The changes look good!

File: src/core/stack.js (hunk 2 above)
Comment (harshithad0703) [reply src/core/stack.js:79]: after making this change, the test are failing

**Review by ishaileshmishra — COMMENTED** [review]
approved

**Pull request merged**